services/src/contexts/migrations/migration_0002_gbif_config.rs

```diff
@@ -0,0 +1,110 @@
+use async_trait::async_trait;
+use tokio_postgres::Transaction;
+
+use crate::error::Result;
+
+use super::database_migration::{DatabaseVersion, Migration};
+
+/// This migration adds the autocomplete timeout field to the GBIF provider config
+pub struct Migration0002GbifConfig;
+
+#[async_trait]
+impl Migration for Migration0002GbifConfig {
+    fn prev_version(&self) -> Option<DatabaseVersion> {
+        Some("0001_raster_stacks".into())
+    }
+
+    fn version(&self) -> DatabaseVersion {
+        "0002_gbif_config".into()
+    }
+
+    async fn migrate(&self, tx: &Transaction<'_>) -> Result<()> {
+        tx.batch_execute(
+            r#"
+                ALTER TYPE "GbifDataProviderDefinition"
+                ADD ATTRIBUTE autocomplete_timeout int
+            "#,
+        )
+        .await?;
+
+        // as ALTER TYPE ADD ATTRIBUTE does not support default values we manually have to set the default value of the new field in the layer_providers table
+        tx.batch_execute(
+            r#"
+                UPDATE layer_providers
+                SET definition.gbif_data_provider_definition.autocomplete_timeout = 3
+                WHERE NOT ((definition).gbif_data_provider_definition IS NULL)
```

Comment:
It also works the way it is right now. It's kind of confusing with composite types.
When you add a new field, it is set to NULL by default (and you can't provide another default). And once ANY field of a composite is NULL, the entire composite is no longer considered NOT NULL, as you say, so you can't check `(definition).gbif_data_provider_definition IS NOT NULL`, it will always be false once any field is NULL. However, once ANY field of a composite is NOT NULL, the composite is not NULL either, so you can use the syntax I used to check whether any field of the composite is NOT NULL.

Reply:
but won't this match all other data provider definitions that are not gbif data providers as well?

Reply:
They have all fields of the `gbif_data_provider_definition` set to NULL, so the whole GBIF definition composite is considered to be NULL